src/contractTest/java/uk/gov/hmcts/reform/sscs/consumer/GetHearingPactConsumerTest.java

```diff
@@ -0,0 +1,454 @@
+    private static final String NOT_FOUND_CASE_ID = "404";
+
+    private static final LocalDateTime date = LocalDateTime.now();
+
+    @Autowired
+    private HmcHearingApi hmcHearingApi;
+
+
+    @Pact(provider = PROVIDER_NAME, consumer = CONSUMER_NAME)
+    public RequestResponsePact getHearing(PactDslWithProvider builder) throws Exception {
+
+        return builder
+            .given("sscs haaring api successfully returns case")
+            .uponReceiving("Request to GET hearing for given valid case ref only")
+            .path(PATH_HEARING)
+            .method(HttpMethod.GET.toString())
+            .query(FIELD_ID + "=" + VALID_CASE_ID)
+            .headers(headers)
+            .willRespondWith()
+            .status(HttpStatus.OK.value())
+            .body(generateValidHearingGetResponsePactDslJsonBody(date))
+            .toPact();
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "getHearing")
+    public void shouldSuccessfullyGetHearing(MockServer mockServer) throws Exception {
+
+        HearingGetResponse result = hmcHearingApi.getHearingRequest(
+            IDAM_OAUTH2_TOKEN,
+            SERVICE_AUTHORIZATION_TOKEN,
+            VALID_CASE_ID
+        );
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new JavaTimeModule());
+        objectMapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
+        HearingGetResponse expected = objectMapper.readValue(
+            generateValidHearingGetResponsePactDslJsonBody(date).toString(),
+            HearingGetResponse.class
+        );
+
+        Assertions.assertEquals(expected, result);
+
+        TimeUnit.SECONDS.sleep(2);
+    }
+
+
+    @Pact(provider = PROVIDER_NAME, consumer = CONSUMER_NAME)
+    public RequestResponsePact getHearingWithRefCheck(PactDslWithProvider builder) throws Exception {
+
+        return builder
+            .given("sscs haaring api successfully returns case")
+            .uponReceiving("Request to GET hearing for given valid case ref only")
+            .path(PATH_HEARING)
+            .method(HttpMethod.GET.toString())
+            .query(FIELD_ID + "=" + VALID_CASE_ID + OPTION_FIELD_IS_VALID)
+            .headers(headers)
+            .willRespondWith()
+            .status(HttpStatus.OK.value())
+            .body(generateValidHearingGetResponsePactDslJsonBody(date))
+            .toPact();
+    }
+
+
+    @Test
+    @PactTestFor(pactMethod = "getHearingWithRefCheck")
+    public void shouldSuccessfullyGetHearingWithRefCheck(MockServer mockServer) throws Exception {
+
+        HearingGetResponse result = hmcHearingApi.getHearingRequest(
+            IDAM_OAUTH2_TOKEN,
+            SERVICE_AUTHORIZATION_TOKEN,
+            VALID_CASE_ID + OPTION_FIELD_IS_VALID
+        );
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new JavaTimeModule());
+        objectMapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
```

Comment:
@dholaramgurjar Did something similar to this but application wide, could you double check with him to make sure everything is in line?

Reply:
this is for the jackson to convert the LocalDateTime correctly, otherwise it will decode it to a string array

Reply:
@dattipoeHMCTS is that ok?